lib/block_view_validator.go

```diff
@@ -16,27 +16,63 @@ import (
+// validator and removes this validator from the set of eligible validators. A user would have to
+// re-register by submitting a subsequent RegisterAsValidator transaction to be re-included.
+//
+// UnjailValidator: Unjails a jailed validator if sufficient time (epochs) have elapsed since the
+// validator was first jailed. A validator is jailed if they fail to participate in consensus by
+// either voting or proposing blocks for too long. A jailed validator is ineligible to receive
+// any block rewards and ineligible to elected leader.
 
 //
 // TYPES: ValidatorEntry
 //
 
 type ValidatorEntry struct {
+	// The ValidatorPKID is the primary key for a ValidatorEntry. It is the PKID
+	// for the transactor who registered the validator. A user's PKID can only
+	// be associated with one validator.
 	ValidatorPKID *PKID
+	// Domains is a slice of web domains where the validator can be reached.
 	// Note: if someone is updating their ValidatorEntry, they need to include
 	// all domains. The Domains field is not appended to. It is overwritten.
-	Domains                  [][]byte
-	DisableDelegatedStake    bool
-	VotingPublicKey          *bls.PublicKey
+	Domains [][]byte
+	// DisableDelegatedStake is a boolean that indicates whether the validator
+	// disallows delegated / 3rd party stake being assigned to themselves. If
+	// a validator sets DisableDelegatedStake to true, then they can still
+	// stake with themselves, but all other users will receive an error if they
+	// try to stake with this validator.
+	DisableDelegatedStake bool
+	// The VotingPublicKey is a BLS PublicKey that is used in consensus messages.
+	// A validator signs consensus messages with their VotingPrivateKey and then
+	// other validators can reliably prove the message came from this validator
+	// by verifying against their VotingPublicKey.
+	VotingPublicKey *bls.PublicKey
+	// The VotingPublicKeySignature is the signature of the SHA256(TransactorPublicKey).
+	// This proves that this validator is indeed the proper owner of the corresponding
+	// VotingPrivateKey.
 	VotingPublicKeySignature *bls.Signature
-	TotalStakeAmountNanos    *uint256.Int
-	RegisteredAtBlockHeight  uint64
-	LastActiveAtEpochNumber  uint64
-	JailedAtEpochNumber      uint64
-	ExtraData                map[string][]byte
-	isDeleted                bool
+	// TotalStakeAmountNanos is a cached value of this validator's total stake, calculated
+	// by summing all the corresponding StakeEntries assigned to this validator. We cache
+	// the value here to avoid the O(N) operation of recomputing when determining a
+	// validator's total stake. This way it is an O(1) operation instead.
+	TotalStakeAmountNanos *uint256.Int
+	// LastActiveAtEpochNumber is the last epoch in which this validator either 1) participated in
+	// consensus by voting or proposing blocks, or 2) unjailed themselves. If a validator is
+	// inactive for too long, then they are jailed.
```

Comment:
Interesting. So is the plan to update this value for all validators who vote in every block every time they vote? Not crazy, though it is O(NumValidators) I guess.